crates/starknet-server/tests/test_dump_and_load.rs

```diff
@@ -217,4 +247,171 @@ mod dump_and_load_tests {
+
+        #[cfg(unix)]
+        {
+            let mut kill = Command::new("kill")
+                .args(["-s", "SIGINT", &devnet_dump.process.id().to_string()])
+                .spawn()
+                .unwrap();
+            kill.wait().unwrap();
+        }
+
+        // file should not be created if there are no transactions
+        if Path::new(dump_file_name).exists() {
+            panic!(
+                "Could find the dump file but there were no transactions to dump {}",
+                dump_file_name
+            );
+        }
+    }
+
+    #[tokio::test]
+    async fn dump_endpoint_fail_with_wrong_request() {
+        let devnet_dump = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+        let dump_body = Body::from(
+            json!({
+                "test": ""
+            })
+            .to_string(),
+        );
+        let result = devnet_dump.post_json("/dump".into(), dump_body).await.unwrap();
+        assert_eq!(result.status(), 422);
+    }
+
+    #[tokio::test]
+    async fn load_endpoint_fail_with_wrong_request() {
+        let devnet_load = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+        let load_body = Body::from(
+            json!({
+                "test": ""
+            })
+            .to_string(),
+        );
+        let result = devnet_load.post_json("/load".into(), load_body).await.unwrap();
+        assert_eq!(result.status(), 422);
+    }
+
+    #[tokio::test]
+    async fn dump_endpoint_fail_with_wrong_file_name() {
+        let devnet_dump = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+        let dump_body = Body::from(
+            json!({
+                "path": "///"
+            })
+            .to_string(),
+        );
+        let result = devnet_dump.post_json("/dump".into(), dump_body).await.unwrap();
+        assert_eq!(result.status(), 500);
+    }
+
+    #[tokio::test]
+    async fn load_endpoint_fail_with_wrong_path() {
+        let load_file_name = "load_file_name";
+        let devnet_load = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+        let load_body = Body::from(
+            json!({
+                "path": load_file_name
+            })
+            .to_string(),
+        );
+        let result = devnet_load.post_json("/load".into(), load_body).await.unwrap();
+        assert_eq!(result.status(), 400);
+    }
+
+    #[tokio::test]
+    async fn dump_load_endpoints_transaction_and_state_after_load_is_valid() {
+        let dump_file_name = "dump_endpoint";
+        let dump_file_name_custom_path = "dump_endpoint_custom_path";
+        let devnet_dump =
+            BackgroundDevnet::spawn_with_additional_args(&["--dump-path", dump_file_name])
+                .await
+                .expect("Could not start Devnet");
+        let mint_tx_hash = devnet_dump.mint(DUMMY_ADDRESS, DUMMY_AMOUNT).await;
+        let dump_body = Body::from(
+            json!({
+                "path": ""
```

Comment:
Ok, so it works if path is an empty string. But what if it's omitted - didn't we allow that in devnet-py?

Reply:
If it's omitted it will not pass JSON validation, but I'll check this case.

Reply:
I think we should either accept the omitted property, or not accept empty strings at all.

Reply:
Fixed in 04ca6f2e8a6b18388572e86e6adddf68f72a3f5b and 60bf67a4a136d6709707914f0aa5c5ab7899893e

Reply:
Now if json body is empty or path is "" we will use default path.

Reply:
"not accept empty strings at all" Why is that? I think that an empty path is straightforward with the intention of using a default path.

Reply:
Straightforward would mean explicit, but here it's implicit (if empty string - use default). A user passing an empty string could also mean that they accidentally parsed something wrong. But I'm ok with keeping this implication if that's how it worked in devnet-py.

Reply:
https://github.com/0xSpaceShard/starknet-devnet/blob/e477aa1bbe2348ba92af2a69c32d2eef2579d863/starknet_devnet/blueprints/base.py#L86

Reply:
It's ok. In devnet-py the defaulting was done if either it was an empty string or if it was `None` (i.e. undefined). Resolving this.